File: algorithms/1796.second-largest-digit-in-a-string/solution.rs
impl Solution {
    pub fn second_highest(s: String) -> i32 {
        let digits = "0123456789";
        let mut nums: Vec<i32> = vec![];
        for c in s.chars() {
            if digits.contains(c) {
                nums.push(c as i32 - 48);
            }
        }
        nums.sort_unstable();
        nums.dedup();

        if nums.len() > 1 {
            nums[nums.len() - 2]
        } else {
            -1
        }
    }
}

File: algorithms/1796.second-largest-digit-in-a-string/solution.rb
# @param {String} s
# @return {Integer}
def second_highest(s)
  digits = ('0'..'9').to_a
  nums = s.chars.select { |c| digits.include?(c) }.sort.uniq
  nums.size > 1 ? nums[-2].to_i : -1
end
